src/System.Runtime.Numerics/src/System/Numerics/Complex.cs

```diff
@@ -295,7 +313,90 @@ public static Complex Exp(Complex value)
+                //   x = \sqrt{\frac{\sqrt{a^2 + b^2} + a}{2}}  y = \pm \sqrt{\frac{\sqrt{a^2 + b^2} - a}{2}}
+                // There is just one complication: depending on the sign on a, either x or y suffers from
+                // cancelation when |b| << |a|. We can get aroud this by noting that our formulas imply
+                // x^2 y^2 = b^2 / 4, so |x| |y| = |b| / 2. So after computing the one that doesn't suffer
+                // from cancelation, we can compute the other with just a division. This is basically just
+                // the right way to evaluate the quadratic formula without cancelation.
+
+                // All this reduces our total cost to two sqrts and a few flops, and it respects the desired
+                // symmetries. Much better than atan + cos + sin!
+
+                // The signs are a matter of choice of branch cut, which is traditionally taken so x > 0 and sign(y) = sign(b).
+      
+                // If the components are too large, Hypot will overflow, even though the subsequent sqrt would
+                // make the result representable. To avoid this, we re-scale (by exact powers of 2 for accuracy)
+                // when we encounter very large components to avoid intermediate infinities.
+                bool rescale = false;
+                if ((Math.Abs(value._real) >= s_sqrtRescaleThreshold) || (Math.Abs(value._imaginary) >= s_sqrtRescaleThreshold))
+                {
+                    if (Double.IsInfinity(value._imaginary) && !Double.IsNaN(value._real))
+                    {
+                        // We need to handle infinite imaginary parts specially because otherwise
+                        // our formulas below produce inf/inf = NaN. The NaN test is necessary
+                        // so that we return NaN rather than (+inf,inf) for (NaN,inf).
+                        return (new Complex(Double.PositiveInfinity, value._imaginary));
+                    }
+                    else
+                    {
+                        value._real *= 0.25;
+                        value._imaginary *= 0.25;
+                        rescale = true;
+                    }
+                }
+ 
+                // This is the core of the algorithm. Everything else is special case handling.
+                double x, y;
+                if (value._real >= 0.0)
+                {
+                    x = Math.Sqrt((Hypot(value._real, value._imaginary) + value._real) * 0.5);
+                    y = value._imaginary / (2.0 * x);
+                }
+                else
+                {
+                    y = Math.Sqrt((Hypot(value._real, value._imaginary) - value._real) * 0.5);
+                    if (value._imaginary < 0.0) y = -y;
+                    x = value._imaginary / (2.0 * y);
+                }
+
+                if (rescale)
+                {
+                    x *= 2.0;
+                    y *= 2.0;
+                }
```

Comment:
@JeffreySax: Basically, this just re-scales by default and tests for underflow afterward, instead of testing for overflow before and re-scaling if required. It's actually a cool idea, but for the moment I'm trying to cross the finish line with this PR and will stick with what mellinoe has approved for now. Please don't take this as a dismissal. I really appreciate your engagement and pointers to standards. I plan to keep working in the code base. I'd appreciate your continued reviews and might well re-consider this when I have more energy for small tweaks.